### PR TITLE
Disable notification option for alerts if notification plugin is not present

### LINKED ISSUE #134: [FEATURE] Disable notification option for alerts if alerting plugin is not present
**Is your feature request related to a problem?**
Creating notification alert option is visible in following flows even if there is no alerting plugin installed.
- [x] Create detector flow
- [x] Edit detector - alert triggers
- [x] Create Alert on Findings

**Do you have any additional context?**
Follow what the alerting plugin does

### PR DESCRIPTION
**Pull request by djindjic**
### Description
Not showing option to create notification if there is no plugin support

### Issues Resolved
Resolves #134 

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/NotificationsCallOut/NotificationsCallOut.tsx
Comment (amsiglan) [inline public/components/NotificationsCallOut/NotificationsCallOut.tsx:15]: Should we use `&nbsp;` instead?

Comment (djindjic) [reply public/components/NotificationsCallOut/NotificationsCallOut.tsx]: Good point, just used the same component from alerting.

Fixed

File: public/components/NotificationsCallOut/NotificationsCallOut.tsx
Comment (djindjic) [reply public/components/NotificationsCallOut/NotificationsCallOut.tsx]: Right.

Fixed

File: public/pages/CreateDetector/components/ConfigureAlerts/components/AlertCondition/AlertConditionPanel.tsx
Comment (amsiglan) [inline public/pages/CreateDetector/components/ConfigureAlerts/components/AlertCondition/AlertConditionPanel.tsx:444]: Should we add the spacer to the callout component itself?

File: public/components/NotificationsCallOut/NotificationsCallOut.tsx
Comment (djindjic) [reply public/components/NotificationsCallOut/NotificationsCallOut.tsx:17]: Done, but just let me know if this is good format
image

File: public/pages/CreateDetector/containers/CreateDetector.tsx
Comment (amsiglan) [inline public/pages/CreateDetector/containers/CreateDetector.tsx:209]: We shouldn't log errors in console. We might have some left in other places but we will clean those up as well.

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (amsiglan) [inline public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:135]: This is same method as above, let's make this a utility function in a helper file and use that?

File: public/pages/Findings/containers/Findings/Findings.tsx
Comment (amsiglan) [inline public/pages/Findings/containers/Findings/Findings.tsx:220]: Same as above, let's make this a shared utility function

File: public/pages/CreateDetector/containers/CreateDetector.tsx
Comment (djindjic) [reply public/pages/CreateDetector/containers/CreateDetector.tsx:209]: OK, right, I found your newer convention.
I have changed this logs to `errorNotificationToast` helper

File: public/pages/CreateDetector/components/ConfigureAlerts/components/AlertCondition/AlertConditionPanel.tsx
Comment (djindjic) [reply public/pages/CreateDetector/components/ConfigureAlerts/components/AlertCondition/AlertConditionPanel.tsx:444]: We can move it it's of course very easy, but personally I wouldn't.

This is not very important case, but its very good example what I was thinking few times in the past.
So, `<NotificationsCallOut>` should not be aware of the parent layout logic (means it should be able to "stretch" as much as parent want, and, generally speaking, should not add outer margins and leave it to the parent too).

I've lately started thinking of it as single responsibility principle no matter its just css/layout-ing. Let me know if you think its more clean way to move it down, no prob.

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (djindjic) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:135]: Yes, I was also thinking on it.

It is the same function on both places (here and in `Findings.tsx`), but result is written to the component state object which I would not force to be named in the same way (`{...plugins: string[];}`).

There are 2 points worth mentioning/discussing:

1. This could be interesting if we would go with functional components and create an custom hook like this:
  ```
  import { usePlugins } from "helperHooks";
  const { plugins } = usePlugins();
  ```
  of corse we would need to think more on different use cases, but generally it could be more "react team" practices aligned. This is just initial point to talk, I'm also not hard opined on this topic yet.

2. I saw in other plugin (index management) too, that frontend services are forwarding whole response to the "consumer", and then consumer (in this case `<UpdateAlertConditions>`) deals with `response.ok` flag. I guess there is a reason for this, but I still haven't got it. I would move that `if(response.ok)` to the service and throw and "native" js error in the else branch. Consumer would just have try/catch around the service call. Ex:
```
// service.ts
...
getPlugins = async (): Promise<ServerResponse<Plugin[]>> => {
 let url = `..${API.PLUGINS}/`;
 const responseObject = await this.httpClient.get(url);

 if (resposne.ok) {
   return responseObject.response;
 } else {
    throw resposne.error;
 }
};
...

// component.tsx
...
try {
  const plugins = await opensearchService.getPlugins();
  this.setState({ plugins: plugins.map((plugin) => plugin.component) });
} catch (e) {
  errorNotificationToast(notifications, 'retrieve', 'plugins', e);
}
...
```

Also, not hardly opined on this just adding this discussion as a contribution.

Let me know if you had some other idea how to move this to the helper by current convenience.

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (AWSHurneyt) [inline public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:204]: In my opinion, it might make more sense to pass the list of installed plugins (i.e., a `string[]`) through the props than a `hasNotificationPlugin` boolean. The reason being that, like security analytics, notifications has a back and frontend plugin. Users are not required to have both installed, so it's possible for just the backend to be installed without the frontend and vise versa. In such a situation, we would want the [channel selection combo box](https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/136/files#diff-fb4d9e38297485bb3a31b9a63e72a9833848dc7b160b65640a7abe85077cf7f6R415) to function, but the [manage channels button](https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/136/files#diff-fb4d9e38297485bb3a31b9a63e72a9833848dc7b160b65640a7abe85077cf7f6R435) would redirect to a non-existent page.

**Comment by AWSHurneyt** [comment]
Could you rebase your dev branch? A PR was merged earlier today which addresses the `Plugin \"securityAnalyticsDashboards\" is only compatible with OpenSearch Dashboards version \"2.4.0\", but used OpenSearch Dashboards version is \"2.4.1\"` error that's impacting the cypress test workflow.

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (djindjic) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:204]: @AWSHurneyt yap we found that edge case of missing frontend plugin, but as I understood in discussion with @amsiglan we ignore it since that would be already "power user" whose knows what he is doing.

If this is not the case, let me know how we can check if frontend plugin is installed (I've researched on it, but haven't found the way) and then we could have all edge cases covered. 

Just for cleaner picture the moment we have 4 cases working in this way:
1. Alerting plugin: Backend 🔴 , Frontend 🔴 -> Red notification **shown** , combobox **disabled**, button **disabled**
3. Alerting plugin: Backend 🔴 ,Frontend 🟢 -> Red notification **shown** , combobox **disabled**, button **disabled**
4. Alerting plugin: Backend 🟢  ,Frontend 🟢 -> Red notification **not shown**, combobox **enabled**, button **enabled**
5. Alerting plugin: Backend 🟢  ,Frontend 🔴  -> Red notification **not shown**, combobox **enabled**, button **enabled + redirect to 404**

P.S. I will change to passing installedPlugins instead of hasNotificationPlugin, but do not understand how to fix missing FE plugin.

File: public/pages/CreateDetector/components/ConfigureAlerts/components/AlertCondition/AlertConditionPanel.tsx
Comment (amsiglan) [reply public/pages/CreateDetector/components/ConfigureAlerts/components/AlertCondition/AlertConditionPanel.tsx:444]: As discussed offline, we can leave it as is

File: public/pages/CreateDetector/containers/CreateDetector.tsx
Comment (amsiglan) [reply public/pages/CreateDetector/containers/CreateDetector.tsx]: As discussed, let's use "console.warn" instead for such errors that are not catastrophic failures for the app. The consumers of this plugin can decide to replace these with some analytics api calls of their own if they wish to monitor logs

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (amsiglan) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:135]: As discussed, let's extract out the common method and have it return the plugins array (which will be empty in case of error). I will refactor it to abstract out the check for "ok" in the service files itself. Thank you for that suggestion

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (amsiglan) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:204]: @AWSHurneyt Do you know of a way to check whether the frontend plugin is installed? If not, I can look into it. I wouldn't block this PR on that though, so we can keep the current behavior where we enable/disable the "combo-box/manage-btn" pair based on backend plugin

Comment (AWSHurneyt) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:204]: Ah, I thought `_cat/plugins` returned the installed frontend plugins as well; but it doesn't seem so. Agreed, let's not block for this reason.

Comment (AWSHurneyt) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:204]: I commented on the issue for this task explaining that we can take this up as a follow-up item.

File: public/pages/CreateDetector/containers/CreateDetector.tsx
Comment (djindjic) [reply public/pages/CreateDetector/containers/CreateDetector.tsx]: done

File: public/pages/Findings/containers/Findings/Findings.tsx
Comment (djindjic) [reply public/pages/Findings/containers/Findings/Findings.tsx:221]: done above

**Comment by djindjic** [comment]
@amsiglan @AWSHurneyt I've addressed all open comments

File: public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx
Comment (amsiglan) [reply public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx:135]: Makes sense, will incorporate this feedback in the refactor

**Review by amsiglan — APPROVED** [review]

**Review by phestecAMZN — APPROVED** [review]

**Pull request merged**
